components/nimbus/src/lib.rs

```diff
@@ -701,6 +713,41 @@ impl AvailableRandomizationUnits {
     }
 }
 
+pub struct NimbusTargetingHelper {
+    targeting_context: TargetingAttributes,
+}
+
+impl NimbusTargetingHelper {
+    pub fn eval_jexl(&self, expr: String, value: Option<JsonObject>) -> Result<bool> {
+        let json = value.map(Value::Object);
+        evaluator::jexl_eval(&expr, &self.targeting_context, json)
+    }
+}
+
+type JsonObject = Map<String, Value>;
+
+// This will need to a UniffiCustomTypeConverter once uniffi 0.17
+// is released and adopted by application-services
+#[cfg(feature = "uniffi-bindings")]
+impl UniffiCustomTypeWrapper for JsonObject {
+    type Wrapped = String;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<JsonObject> {
+        let json: Value = serde_json::from_str(&val)?;
+
+        match json.as_object() {
+            Some(obj) => Ok(obj.clone()),
+            _ => Err(uniffi::deps::anyhow::anyhow!(
+                "Unexpected JSON-non-object in the bagging area"
```

Comment:
Hmm can we add tests for a non-object json? to make sure the exception is handled correctly by this - I think the invalid json case is covered by the bindings side custom types right?

Reply:
I've added tests, but we try hard to make sure strings that don't code for objects enter the system.